### DISCUSSION TITLE
dotnet publish inside Dockerfile doesn't work on 8.0-alpine

### DISCUSSION THREAD

**Discussion by rcdailey** [dotnet/dotnet-docker#5333] [Q&A] [answered]
I'm referencing the sample [here]. For my personal project, I have this in the build portion of my `Dockerfile`:

```dockerfile
# syntax=docker/dockerfile:1.7-labs

# Following Microsoft's pattern shown here:
# https://github.com/dotnet/dotnet-docker/blob/main/samples/dotnetapp/Dockerfile.alpine
# https://github.com/dotnet/dotnet-docker/blob/main/samples/README.md
#
FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/sdk:8.0-alpine AS build

ARG TARGETARCH
WORKDIR /source

# copy csproj and restore as distinct layers
COPY --parents *.props src/*/*.csproj ./
RUN dotnet restore src/Recyclarr.Cli -a $TARGETARCH

# copy and publish app and libraries
COPY . .
RUN dotnet publish src/Recyclarr.Cli -a $TARGETARCH --no-restore -o /app
```

You can find this and more in my open source repo [here](https://github.com/recyclarr/recyclarr/compare/master...build-inside-docker#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557).

When I run `docker compose build`, I get this error message during the `dotnet publish` step:

> error NETSDK1047: Assets file '/source/src/Recyclarr.Cli/obj/project.assets.json' doesn't have a target for 'net8.0/linux-musl-x64'. Ensure that restore has run and that you have included 'net8.0' in the TargetFrameworks for your project. You may also need to include 'linux-musl-x64' in your project's RuntimeIdentifiers.

As far as I can tell, I'm doing exactly what the sample is doing. I also added the `-v d` argument to the `dotnet restore` command to see more verbose output. It does indeed show `linux-musl-x64` being used to restore packages.

I'm completely lost here. What am I doing wrong?

**Comment by MichaelSimons** [comment, marked as answer]
Do you have a .dockerignore?  The `COPY . .` can be interfering if you don't and you have built locally.  From https://learn.microsoft.com/en-us/dotnet/core/tools/sdk-errors/netsdk1005

> If you're building a Docker image, make sure the .dockerignore file ignores the bin and obj directories. For more information, see GitHub pull request [dotnet/docs #29530](https://github.com/dotnet/docs/pull/29530).

**Reply by rcdailey** [reply]
I do not have a `.dockerignore`, so that's most likely the issue. I feel like `.dockerignore` is redundant when I have `.gitignore`; very frustrating that I have to add it. I do appreciate your help.

**Reply by rcdailey** [reply]
I tested this and indeed it does work. The `.dockerignore` was an oversight on my part, mainly because I was hyper-focused on the `README.md`. I think it might help to mention the importance of the `.dockerignore` in the readme itself. [This one](https://github.com/dotnet/dotnet-docker/blob/main/samples/dotnetapp/README.md), for example, doesn't mention it at all. A warning-style admonition to make something like this catch people's eye would have helped me, at least!

Thank you again for your help.

**Reply by MichaelSimons** [reply]
Please feel free to open an issue and/or PR to help improve the READMEs.

**Comment by richlander** [comment]
TIL https://docs.docker.com/reference/dockerfile/#copy---parents

**Reply by rcdailey** [reply]
Yep, pretty essential in the complex app example they have.
